### PR TITLE
Add a config writing method to Halibot core

### PR DESCRIPTION
**Pull request by richteer**
In `main.py`, we use the `Halibot` class to load the config, so we might as well write the config using it as well. We should probably consider a kwarg to `Halibot` for `config_path`, so we can stop hardcoding `"config.json"` everywhere -- it can be a command line argument if people really want to change it. Or if we decide that it should be `"halibot.json"` (which I'm starting to think might be better for clarity), it would be easier to fix everywhere.

Also included is a patch that cleans up some trailing whitespace. Figured it didn't need its own PR.

### CODE REVIEW

**Comment by richteer** [comment]
I can code good. Fixed my stupid bugs from before, tests now pass.

**Comment by richteer** [comment]
Okay, I think I actually fixed the stupid bugs this time. I no code good.

File: halibot/halibot.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def _load_config(self):
 			self.config = json.loads(f.read())
 			halibot.packages.__path__ = self.config.get("package-path", [])
 
+	def _write_config(self):
+		with open("config.json", "w") as f:
+			f.write(json.dumps(self.config, sort_keys=True, indent=4))
+
+
```
Comment (sjrct) [inline halibot/halibot.py]: Extra newline.

**Review by sjrct — CHANGES_REQUESTED** [review]
Missed writing the config again at `main.py#43`. Might want to be able to take a file path as an argument to accommodate this line, or maybe make an active directory a field of the `Halibot` object and have it load/write the config from there. I favor the second approach myself.

I also think we should remove the check for the config's existence at `main.py#59`. Probably would be better to fail in `_load_config`. And with the second approach to the config location from above, we do not have to worry about this getting off kilter when running with a non-local config.

**Comment by richteer** [comment]
> Missed writing the config again at main.py#43

Yeah, I was purposely ignoring those since they were effectively dumping a new file, rather than writing a `Halibot` instance's config. I agree the "active directory" field is probably the better approach in general. 

What are your thoughts on implementation? Default value to `"."`? kwarg on `__init__()`? Or just a straight field that can be modified? Personally, I'm partial to putting it on init, with a default of `"."`.

Side note: I'm less worried about `main.py`, this was more a move for `main.py`'s eventual successor to call into. I was also writing some auto module-config updating logic that used this, but that got more complicated than it was worth.

**Comment by sjrct** [comment]
I agree main.py is not something to worry about too much. I am also partial to `def __init__(..., working_directory=".")` or something like that.

**Review by sjrct — APPROVED** [review]

**Review by sjrct — APPROVED** [review]

**Pull request merged**